src/Raven.Server/Documents/Indexes/MapReduce/Auto/AutoMapReduceIndexDefinition.cs

```diff
@@ -19,13 +19,17 @@ internal class AutoMapReduceIndexDefinition : AutoIndexDefinitionBaseServerSide
 
         public readonly AutoIndexField[] OrderedGroupByFields;
 
+        public List<string> GroupByFieldNames { get; }
+
         public AutoMapReduceIndexDefinition(string collection, AutoIndexField[] mapFields, AutoIndexField[] groupByFields, IndexDeploymentMode? deploymentMode, IndexDefinitionClusterState clusterState, long? indexVersion = null)
             : base(AutoIndexNameFinder.FindMapReduceIndexName(collection, mapFields, groupByFields), collection, mapFields, deploymentMode, clusterState, indexVersion)
         {
             OrderedGroupByFields = groupByFields.OrderBy(x => x.Name, StringComparer.Ordinal).ToArray();
 
             GroupByFields = groupByFields.ToDictionary(x => x.Name, x => x, StringComparer.Ordinal);
 
+            GroupByFieldNames = groupByFields.Select(x => x.Name).ToList();
```

Comment:
I'm wondering if this is not safer to use here OrderedGroupByFields instead of groupByFields